python/pyarrow/table.pxi

```diff
@@ -6259,6 +6259,56 @@ def concat_tables(tables, MemoryPool memory_pool=None, str promote_options="none
     return pyarrow_wrap_table(c_result_table)
 
 
+def concat_recordbatches(recordbatches, MemoryPool memory_pool=None):
+    """
+    Concatenate pyarrow.RecordBatch objects.
+
+    All recordbatches must share the same Schema,
+    the operation is guaranteed to be zero-copy.
```

Comment:
I suppose this method would never be zero copy? (the result is again a RecordBatch, so it actually needs to copy to concatenate the arrays)

Reply:
Ops, you are right, I don't know why I was thinking of creating a RecordBatch from a set of arrays and not about extending the arrays themselves.

Reply:
fixed